sympy/solvers/decompogen.py

```diff
@@ -51,6 +53,16 @@ def decompogen(f, symbol):
         result += [f.subs(arg, symbol)] + decompogen(arg, symbol)
         return result
 
+    # ===== Min/Max Functions ===== #
+    if isinstance(f, (Min, Max)):
+        if And(*[a.has(symbol) for a in f.args]):
+            raise TypeError('cannot decompose %s' % f)
+        for i in f.args:
+            if i.has(symbol):
+                arg = i
```

Comment:
`arg` is not used below; what is the purpose of this loop? The result now is only that `i = f.args[-1]`.

Reply:
I used `i` wrongly below. It will be `arg` below.